### ISSUE TITLE
[gtk3] Binding of show/hide hotkey fails on Wayland

### ISSUE THREAD

**Issue by coder-hugo** [Guake/guake#1109]
I just upgraded to version 3.0.2 of guake which uses GTK3 instead of GTK2. As a result of this guake runs with full Wayland support. With Wayland there is no way for an application to register a global hotkey like it's possible with X11. My current workaround to get the new version of guake running under Wayland is to set the following environement variable `GDK_BACKEND=x11` for the guake process. With this guake 3 will use also XWayland like it was the case for guake 2. So for now it's working but I think the better approach to solve this issue is to create a GAction which would appear in the global hotkeys section of the compositor. See [this issus](https://github.com/kupferlauncher/keybinder/issues/11#issuecomment-333390043) for more information.

Edit: I just figured out that the workaround with the `GDK_BACKEND` environment variable isn't the best. In this case the show/hide hotkey works only if the window that has currently the focus is also rendered by XWayland. So I looked for a better workaround and found the command line flag `--toggle-visibility`. So I configured a global hotkey in my compositor that calls `guake --toggle-visibility`. With this there is no need to run guake using the x11 backend. But to get this working the changes of the following commit 0f2da79c48cee706d389413a7e9c7ae715cb7ef6 are required which aren't yet contained in a release.

**Comment by coder-hugo** [comment]
Just saw the following issue #1084 that describes the same issue. Unfortunately I haven't found it during my searches before I opened this separate issue. So one of these issues can be closed.

**Comment by gsemet** [comment]
Hi @coder-hugo. I understand your point, but I need to find a solution that satisfy the most people. I got the packaging of Guake blocked on Fedora because the global hotkey does not work, and I receive several new issue every week about Guake not working on Wayland. So for the moment, if the X11 backend works, I'll stick with it.

I cannot develop under wayland, I don't use it on any of my system, sadly. So, feel free to hack into the code and propose a pull request. I have opened a ticket to add this support on #1343.

**Comment by coder-hugo** [comment]
Hi @gsemet. I also understand your point. But in this case I would definitely prefer to have an option for forcing the X11 backend. To prevent all those new issues of course this should be set to true by default. And the UI for this option should show a hint that disabling it may cause some issues.
The reason why I'm not happy with this change is that this is a step backwards for the next version of Guake. It's not just that the scaling doesn't work properly with the X11 backend. The X11 is also a very insecure protocol compared with wayland. With wayland an application can only access the content of its own windows. This is not the case for X11. Under X11 each application can access the content of other windows and also catch events that happened outside of its window(s). I'm pretty happy that currently more and more of my day-to-day used apps get adapted to support wayland so that losing this support for Guake wouldn't be that cool.
Maybe I find some time at the weekend to open a pull request at least to have an option to disable the forcing of the X11 backend.

**Comment by gsemet** [comment]
i'll be thinking into adding the "X11 compatibility mode for wayland" option, but currently there is no way to have libkeybinder work under wayland. So user needs to hack their own guake shortcut on GNOME, and kde user has to make the same, and so on. If i can get help on this maybe guake can be smoothly integrated and future-proof, but most of the time, I cannot do it alone, not having a Wayland nor a KDE or other than Gnome desktop.
